### PR TITLE
Update Actions section to reference ehrQL

### PR DESCRIPTION
**Pull request by inglesp**
Addresses part of https://github.com/opensafely-core/ehrql/issues/1676

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -42,7 +42,6 @@ nav:
       - Actions:
           - Overview: actions-intro.md
           - The project pipeline: actions-pipelines.md
-          - The cohortextractor action: actions-cohortextractor.md
```
Comment (StevenMaude) [inline mkdocs.yml:45]: question: Is this page to be removed? Or is the intent just to remove it from the navigation alone (which is fine)?

File: docs/actions-pipelines.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ This section covers how to develop, run, and test your code to ensure it will wo
 
 ## Project pipelines
 
-The [cohortextractor](actions-cohortextractor.md) section describes how to make an action which generate dummy datasets based on the instructions [defined in your `study_definition.py` script](study-def.md).
+The [ehrQL](/ehrql/) documentation describes how to make an action which generate dummy datasets based on the instructions defined in your `dataset_definition.py` script.
```
Comment (StevenMaude) [inline docs/actions-pipelines.md:8]: suggestion: Maybe linking to the dummy data page for ehrQL dataset definitions would be better?

```suggestion
The [ehrQL](/ehrql/how-to/dummy-data.md) documentation describes how to make an action which generate dummy datasets based on the instructions defined in your `dataset_definition.py` script.
```

(I've not checked the correctness of that relative link, but the page is better than linking to a more general part of the ehrQL documentation, I think.)

Comment (StevenMaude) [inline docs/actions-pipelines.md:8]: thought: There's also a slight quirk here where, previously, `study_definition.py` encompassed measures or what we now call datasets, whereas `dataset_definition.py` explicitly refers to datasets (and presumably someone would use `measure_definition.py` instead for measures).

I don't think probably matters too much though.

**Review by StevenMaude — APPROVED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (inglesp) [reply mkdocs.yml:45]: I'm going to salvage some possibly-useful cohortextractor stuff for some legacy docs.

Comment (StevenMaude) [reply mkdocs.yml:45]: Great :+1:

File: docs/actions-pipelines.md (hunk 2 above)
Comment (inglesp) [reply docs/actions-pipelines.md]: I don't think these docs are tight enough to make this worth worrying about.

**Pull request merged**